WordPress/gutenberg
# Add indicator for metadata changes to Save Panel when reviewing modified entities

## What?

This is a first iteration of adding metadata to the Save Panel when reviewing modified entities.

## Why?

Begins addressing https://github.com/WordPress/gutenberg/issues/61405
Related to https://github.com/WordPress/gutenberg/pull/61753
We want to indicate to the user when they are modifying the value of custom fields.

## How?

Overrides existing logic to determine whether to show the Save Panel, while also showing that metadata has changed.

## Testing Instructions




<details>
<summary>1. Register post meta by adding this snippet to your theme's functions.php</summary>

```php
add_action( 'init', 'test_block_bindings' );

function test_block_bindings() {
	register_meta(
		'post',
		'text_field',
		array(
			'show_in_rest'      => true,
			'single'            => true,
			'type'              => 'string',
			'default'           => 'default text value',
		)
	);
}
```

</details>


<details>

<summary>2. Add a paragraph block bound to the custom field using the Code Editor</summary>

```

<p>Paragraph content</p>

```

</details>
 
3. Add a normal paragraph block and modify it.
4. Run the following code in your browser console:
```
const meta = wp.data.select('core/editor').getEditedPostAttribute('meta');
meta['text_field'] = 'testing change';
wp.data.dispatch('core/editor').editPost( meta );
```
5. Press the Save button.
6. Verify that the Save Panel appears and shows that metadata has changed.

### Testing Instructions for Keyboard


## Screenshots or screencast 

<img width="889" alt="333905890-a40e6c2c-7918-42ce-9f6b-69047fe86971" src="https://github.com/WordPress/gutenberg/assets/5360536/fb63ad5f-6412-48fe-88c7-0835b4deb2b9">
